### PR TITLE
Clarify the Box layout diagrams.

### PR DESCRIPTION
**Pull request by gendx**
At first it wasn't clear to me what "tag", "0" and "1" were supposed to represent. I initially read the diagram as "the tag is two bits, 01".

It's clearer with a header for each column.

### CODE REVIEW

**Comment by mgeisler** [comment]
> It's clearer with a header for each column.

Yes, kind of... except that this is how the diagrams looked at first until I re-drew them to remove clutter the other diagrams :smile: 

I would love it if we could find an even better representation... perhaps we could drop the idea of the field names and just show the payloads instead. Something like this (made with https://asciiflow.com):

```
+----+----+--+  +----+----+----+  +----+---------+
|Cons| 1  |  +->|Cons| 2  |    |->|Nil |/////////|
+----+----+--+  +----+----+----+  +----+---------+
```

That is a nice symbolic representation and it should be possible to change this into a drawing where the pointer shares the space with the `Cons`/`Nil` tags.

**Comment by gendx** [comment]
Applied the suggestion.

File: src/std/box-niche.md [hunk 1]
```diff
@@ -18,14 +18,15 @@ allows the compiler to optimize the memory layout:
...
-:   | "1/Tag"| o-----+----+-----+-'  | "1/Tag"|  o-----+-'  | "1/Tag"| null |   :
-:   +--------+-------+    :     :    +--------+--------+    +--------+------+   :
-:                         :     :                                               :
-:                         :     :                                               :
-`- - - - - - - - - - - - -'     '- - - - - - - - - - - - - - - - - - - - - - - -'
+.- - - - - - - - - - - -.     .- - - - - - - - - - - - - - - - - - - - - - -.
+:                       :     :                                             :
+:    list               :     :                                             :
+:   +----+----------+   :     :    +----+----------+    +----+----------+   :
+:   | .0 | ".1/Tag" |   :     :    | .0 | ".1/Tag" |    | .0 | ".1/Tag" |   :
```
Comment (mgeisler) [inline src/std/box-niche.md:25]:
```suggestion
```

File: src/std/box-niche.md [hunk 2]
```diff
@@ -18,14 +18,15 @@ allows the compiler to optimize the memory layout:
...
-:   | 0      | 1     |    :     : .->| 0      |  2     | .->| ////// | //// |   :
-:   | "1/Tag"| o-----+----+-----+-'  | "1/Tag"|  o-----+-'  | "1/Tag"| null |   :
-:   +--------+-------+    :     :    +--------+--------+    +--------+------+   :
-:                         :     :                                               :
-:                         :     :                                               :
-`- - - - - - - - - - - - -'     '- - - - - - - - - - - - - - - - - - - - - - - -'
+.- - - - - - - - - - - -.     .- - - - - - - - - - - - - - - - - - - - - - -.
+:                       :     :                                             :
+:    list               :     :                                             :
+:   +----+----------+   :     :    +----+----------+    +----+----------+   :
```
Comment (mgeisler) [inline src/std/box-niche.md:24]:
```suggestion
```

File: src/std/box-recursive.md [hunk 3]
```diff
@@ -20,24 +20,24 @@ fn main() {
 .- - - - - - - - - - - - -.     .- - - - - - - - - - - - - - - - - - - - - - - -.
 :                         :     :                                               :
 :    list                 :     :                                               :
-:   +--------+-------+    :     :    +--------+--------+    +--------+------+   :
-:   | Tag    | Cons  |    :     : .->| Tag    | Cons   | .->| Tag    | Nil  |   :
-:   | 0      | 1     |    :     : |  | 0      | 2      | |  | ////// | //// |   :
-:   | 1      | o-----+----+-----+-'  | 1      | o------+-'  | ////// | //// |   :
-:   +--------+-------+    :     :    +--------+--------+    +--------+------+   :
+:   +------+----+----+    :     :    +------+----+----+    +------+----+----+   :
```
Comment (mgeisler) [inline src/std/box-recursive.md:23]:
```suggestion
```

File: src/std/box-recursive.md [hunk 4]
```diff
@@ -20,24 +20,24 @@ fn main() {
...
+:   | Tag  | .0 | .1 |    :     :    | Tag  | .0 | .1 |    | Tag  | .0 | .1 |   :
```
Comment (mgeisler) [inline src/std/box-recursive.md:24]:
```suggestion
```

**Review by mgeisler — APPROVED** [review]
Thank you! I'll go a step further and completely remove the confusing `.1/Tag` part.

File: src/std/box-niche.md (hunk 1 above)
Comment (mgeisler) [reply src/std/box-niche.md:25]: Ah, I had hoped I could sneak this in :-) Let me do it in a new PR.

**Pull request merged**